pkg/collect/host_certificate.go

```diff
@@ -77,3 +79,11 @@ func isEncryptedKey(filename string) (bool, error) {
 	}
 	return bytes.Contains(data, []byte("ENCRYPTED")), nil
 }
+
+func (c *CollectHostCertificate) RemoteCollect(progressChan chan<- interface{}) (map[string][]byte, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (c *CollectHostCertificate) HasRemoted() bool {
```

Comment:
If the only purpose of `HasRemote()` is to check that a collector has implemented the `RemoteCollect()` function, I'd leave the function out in favour or checking the `not implemented` error. It would be good to keep public functions at a minimum

You could do something like
```go
var ErrNotImplemented = errors.New("not implemented")
...
result, err := collector.RemoteCollect(opts.ProgressChan)
if errors.Is(err, collect.ErrNotImplemented) {
	result, err = collector.Collect(opts.ProgressChan)
	...
}

if err != nil {
...
}
```

Reply:
good idea, I have changed to 
```
if errors.Is(err, collect.ErrRemoteCollectorNotImplemented) {
```